apps/frontend/src/context/DataContext.tsx

```diff
@@ -139,7 +191,10 @@ export const DataContextProvider = ({ clockWorker, children }: Props) => {
           const deltaDistance = (speed * action.delta) / 1000;
 
           const totalDistance = currentData.distance + deltaDistance;
-          const coordinates = distanceToCoordinates(zap, totalDistance);
+          const coordinates = distanceToCoordinates(
+            route === 'D' ? dRaw : zap,
```

Comment:
these equality checks would probably be cleaner and safer for additional options using a function or maybe a total dict

Reply:
Yeppp! Made a function in `gps.ts`!